### PR TITLE
Add Loki as a log exporter (first PR: overall structure)

### PR DESCRIPTION
**Pull request by gramidt**
**Description:**
Allow exporting of log data to Loki.

**Link to tracking Issue:** 
#1894

**Testing:**
Unit tests are included

**Documentation:**
The documentation includes a short description and the configuration.

### CODE REVIEW

**Comment by gramidt** [comment]
@open-telemetry/collector-contrib-approvers @anuraaga - Looking forward to working with you and getting Loki added as an exporter. Per the contributing docs, the second PR will include the implementation, and the third will add it to the binary.

**Comment by clhain** [comment]
LGTM

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:26]:
```suggestion
- `ca_file` (no default) Path to the CA cert to verify the server being connected to.
```

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:28]:
```suggestion
- `cert_file` (no default) Path to the TLS cert to use for client connections when TLS client auth is required. Should only be used if `insecure`
```

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:30]: Ditto

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:16]: This is so long even the example doesn't use it :) How about just `attributes_for_labels` or similar? Also should these be regexes for more flexibility?

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:48]: These are supposed to be attribute names I think so using realizstic semantic convention names would be clearer.

File: exporter/lokiexporter/loki.go
Comment (anuraaga) [inline exporter/lokiexporter/loki.go:44]: Update this to the actual config name

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md:16]: Absolutely agree on the length and the use of regular expressions. Updating now.

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md]: Addressed.

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md]: Addressed

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md:29]: Addressed

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md]: Addressed

File: exporter/lokiexporter/loki.go
Comment (gramidt) [reply exporter/lokiexporter/loki.go]: Addressed

**Comment by gramidt** [comment]
Thank you so much for reviewing, @anuraaga! I have addressed your comments and look forward to your review.

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:28]: Since these are for client auth, I guess they're independent of the `insecure` flag? `insecure` is to allow TLS even if `ca_file` isn't set I think

File: exporter/lokiexporter/README.md
Comment (anuraaga) [inline exporter/lokiexporter/README.md:3]: Can an exporter require TLS? I guess it's servers that require it, not exporters. Here it will be TLS or not based on whether endpoint is https or http I think. Or if plaintext really isn't supported, then can you add to the docs for `endpoint` that the URL must start with `https://`?

**Review by anuraaga — APPROVED** [review]
Couple more readme nits but LGTM, thanks

**Comment by gramidt** [comment]
Thank you so much, @anuraaga!

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md:28]: That is correct. The `insecure` flag configures the HTTP client to not validate the server certificate, so it wouldn't make sense to provide a certs if validation is disabled. 

Updating docs so that it says "Should only be used if `insecure` is set to false.

File: exporter/lokiexporter/README.md
Comment (gramidt) [reply exporter/lokiexporter/README.md:3]: After a bit more thought, I have removed the requirements for https by default, since some Loki instances my be colocated in the same network as the Otel collector and organizations may not require TLS within the same network.

Updating docs now.

**Comment by gramidt** [comment]
@anuraaga - Are you authorized to merge this?

**Comment by gramidt** [comment]
What are the next steps to getting this merged, @anuraaga?

**Comment by anuraaga** [comment]
@gramidt We need a maintainer to merge this after the initial review. I guess people are on holidays so we may need to wait until next week.

**Comment by gramidt** [comment]
Sounds good! Thank you again for all of your help, @anuraaga!

**Comment by bogdandrutu** [comment]
@gramidt consider to add an entry in the codeowners for this component as well as for the dependabot

**Comment by gramidt** [comment]
Thank you so much for your help in getting this merged, @bogdandrutu! I'll be sure to commit the entrees for the codeowners.

**Pull request merged**
